### DISCUSSION TITLE
APIv4 feature request: allow contributionsCollection to include private contributions

### DISCUSSION THREAD

**Discussion by craigfe** [community/community#24812] [Apps, API and Webhooks]
<p>The GraphQL API contains a contributionsCollection endpoint which is excellent for writing work reports:</p>
<pre><code>query {
  viewer {
    contributionsCollection {
      pullRequestContributionsByRepository {
        repository {
          nameWithOwner
        }
        contributions {
          nodes {
            pullRequest {
              createdAt
              title
              url
            }
          }
        }
      }
}
}
}
 
</code></pre>
<p>To my knowlege, it is impossible to get this query to include private contributions, even when appropriately authenticated (although I can do things like counting the number of contributions that <em>can’t</em> be shown). This is corroborated by <a href="https://github.community/t/graphql-github-v4-api-get-last-activity-for-a-user-in-an-organisation/14415/6">a recent answer on this forum</a>. I feel that this would a really valuable feature to add, if possible.</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/craigfe">@craigfe</a>,</p>
<p>Thank you for being here! As you mentioned this is not possible on v4 yet, I’ll be sure to pass it along to our team for consideration, in the meanwhile  you can use the REST API which allows you to list commits for a specific user:</p>
<p><a href="https://developer.github.com/v3/repos/commits/#list-commits-on-a-repository" rel="nofollow noopener">https://developer.github.com/v3/repos/commits/#list-commits-on-a-repository</a></p>
<p>Sorry I didn’t have better news, but please rest assured your feedback is correct hands.</p>

**Comment by colin-i** [comment]
For private repositories at contributionsCollection, the token must have **read:user** access.

**Comment by gpetiot** [comment]
It seems private repos are still missing from the contributions using the GraphQL API, even with a token having `user:read` access.
